libs/pika/executors/include/pika/executors/thread_pool_scheduler_queue_bypass.hpp

```diff
@@ -0,0 +1,275 @@
+
+    ///////////////////////////////////////////////////////////////////////
+    struct thread_pool_scheduler_queue_bypass : thread_pool_scheduler
+    {
+        using thread_pool_scheduler::get_fallback_annotation;
+        using thread_pool_scheduler::thread_pool_scheduler;
+        using thread_pool_scheduler::operator==;
+        using thread_pool_scheduler::operator!=;
+
+        // Since this scheduler executes the task immediately on whichever thread is active
+        // we do not support "with_priority" property
+        // we do not support "with_hint" property
+
+        // support with_stacksize property
+        friend thread_pool_scheduler_queue_bypass tag_invoke(
+            pika::execution::experimental::with_stacksize_t,
+            thread_pool_scheduler_queue_bypass const& scheduler,
+            pika::execution::thread_stacksize stacksize)
+        {
+            auto sched_with_stacksize = scheduler;
+            sched_with_stacksize.stacksize_ = stacksize;
+            return sched_with_stacksize;
+        }
+
+        friend pika::execution::thread_stacksize tag_invoke(
+            pika::execution::experimental::get_stacksize_t,
+            thread_pool_scheduler_queue_bypass const& scheduler)
+        {
+            return scheduler.stacksize_;
+        }
+
+        // support with_annotation property
+        friend constexpr thread_pool_scheduler_queue_bypass tag_invoke(
+            pika::execution::experimental::with_annotation_t,
+            thread_pool_scheduler_queue_bypass const& scheduler, char const* annotation)
+        {
+            auto sched_with_annotation = scheduler;
+            sched_with_annotation.annotation_ = annotation;
+            return sched_with_annotation;
+        }
+
+        friend thread_pool_scheduler_queue_bypass tag_invoke(
+            pika::execution::experimental::with_annotation_t,
+            thread_pool_scheduler_queue_bypass const& scheduler, std::string annotation)
+        {
+            auto sched_with_annotation = scheduler;
+            sched_with_annotation.annotation_ =
+                pika::detail::store_function_annotation(PIKA_MOVE(annotation));
+            return sched_with_annotation;
+        }
+
+        // support get_annotation property
+        friend constexpr char const* tag_invoke(pika::execution::experimental::get_annotation_t,
+            thread_pool_scheduler_queue_bypass const& scheduler) noexcept
+        {
+            return scheduler.annotation_;
+        }
+
+        template <typename F>
+        void execute(F&& f, char const* fallback_annotation) const
+        {
+            using namespace pika::debug::detail;
+            using namespace pika::threads::detail;
+            using namespace pika::detail;
+
+            if (get_self_id() != invalid_thread_id)
+            {
+                throw std::runtime_error("Bypass scheduler - already on task thread");
+                return;
```

Comment:
```suggestion
                PIKA_THROW_EXCEPTION(pika::error::invalid_status, "thread_pool_scheduler_queue_bypass::execute", "Already on a pika task");
                return;
```
(with appropriate line wrapping)